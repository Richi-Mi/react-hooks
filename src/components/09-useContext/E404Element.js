import React from 'react'

const E404Element = () => {
    return (
        <div>
           <h1> Pagina no existe 404 </h1> 
        </div>
    )
}

export default E404Element
