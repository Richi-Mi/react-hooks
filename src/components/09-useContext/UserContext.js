import { createContext } from 'react';

export const UserContext = createContext( null ); // Creamos un nuevo contexto, funciona como
// componente