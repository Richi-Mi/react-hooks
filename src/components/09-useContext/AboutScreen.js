import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const AboutScreen = () => {
    const { user } = useContext( UserContext ); // Obtenemos una propiedad del Context
    return (
        <div>
            <h1> About App </h1>
            <hr />

            <pre className='container'>
                { JSON.stringify( user, null, 3 )}
            </pre>
        </div>
    )
}

export default AboutScreen
