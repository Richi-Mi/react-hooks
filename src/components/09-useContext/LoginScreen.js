import React, { useContext } from 'react'
import { UserContext } from './UserContext';

const LoginScreen = () => {
    // 1. Obtener la referencia al userContext
    const { setUser, user } = useContext( UserContext );
    const usuario = {
        id: new Date().getTime(),
        ...user
    }
    // 2. setUser 
    return (
        <div>
            <h1> Login Screen </h1>
            <hr />

            <button className='btn btn-primary' onClick={ () => setUser( usuario ) }> Login </button>
            <button className='btn btn-danger' onClick={ () => setUser({}) }> LogOut </button>
        </div>
    )
}

export default LoginScreen
