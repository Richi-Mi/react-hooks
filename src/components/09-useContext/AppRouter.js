import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes
} from "react-router-dom";

import AboutScreen from './AboutScreen';
import E404Element from './E404Element';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavBar from './NavBar';
/**
 * <Router /> .- Solo sirve para definir que esta parte va a ser un Router.
 * <Routes> .- Aqui meteremos todas las Rutas de nuestra SPA 
 * <Route> .- Definimos una nueva ruta de nuestra App. 
 */

const AppRouter = () => {
    return (
        <Router>
            <div >
                <NavBar />
                {/**<Routes></Routes> Nos ayuda a mostrar rutas */}
                <Routes>
                    <Route path="/" element={<HomeScreen />} />

                    <Route path="/about" element={<AboutScreen />} />
                    <Route path="/login" element={<LoginScreen />} />
                    
                    <Route path="*" element={<E404Element />}/>

                </Routes>

            </div>
        </Router>
    )
}

export default AppRouter
