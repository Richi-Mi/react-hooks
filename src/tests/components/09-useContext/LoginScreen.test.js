import { mount } from 'enzyme';
import React from 'react';
import LoginScreen from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe(' * Pruebas en el componente <LoginScreen />', () => {
    const user = {
        name: 'Richi!!',
        email: 'richi@gmail.com'
    }
    const setUser = jest.fn();
    const value = {
        user,
        setUser
    }
    // wrapper // mount
    const wrapper = mount(
        <UserContext.Provider value={ value }>
            <LoginScreen />
        </UserContext.Provider>
    )
    test(' - Debe de mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });
    test(' - Debe de ejecutar el setUser() con el argumento esperado', () => {

        const btn_login = wrapper.find('.btn-primary');
        btn_login.simulate('click');

        expect( setUser ).toHaveBeenCalled();
        expect( setUser ).toHaveBeenCalledTimes(1);
        expect( setUser ).toHaveBeenCalledWith({
            id: expect.any( Number ),
            ...user
        });
    
    });
    
});
