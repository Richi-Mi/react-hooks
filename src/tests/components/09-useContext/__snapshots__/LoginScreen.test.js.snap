// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas en el componente <LoginScreen />  - Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
     Login Screen 
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
     Login 
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
     LogOut 
  </button>
</div>
`;
